diff --git a/bison/floorplan_parser.y b/bison/floorplan_parser.y
index 52eca8a..e29c68d 100644
--- a/bison/floorplan_parser.y
+++ b/bison/floorplan_parser.y
@@ -42,6 +42,7 @@
     #include "floorplan_element.h"
     #include "ic_element.h"
     #include "powers_queue.h"
+    #include "floorplan.h"
 }
 
 %union
@@ -100,9 +101,9 @@
 
 %parse-param { Floorplan_t  *floorplan  }
 %parse-param { Dimensions_t *dimensions }
-%parse-param { yyscan_t      scanner    }
+%parse-param { void         *scanner    }
 
-%lex-param   { yyscan_t scanner       }
+%lex-param   { void *scanner }
 
 %initial-action
 {
diff --git a/bison/stack_description_parser.y b/bison/stack_description_parser.y
index 45977eb..b69072f 100644
--- a/bison/stack_description_parser.y
+++ b/bison/stack_description_parser.y
@@ -44,6 +44,9 @@
     #include "die.h"
     #include "stack_element.h"
     #include "inspection_point.h"
+    #include "analysis.h"
+    #include "output.h"
+    #include "stack_description.h"
 }
 
 // The YYSTYPE union used to collect the types of tokens and rules
@@ -201,9 +204,9 @@
 %parse-param { StackDescription_t *stkd     }
 %parse-param { Analysis_t         *analysis }
 %parse-param { Output_t           *output   }
-%parse-param { yyscan_t            scanner  }
+%parse-param { void               *scanner  }
 
-%lex-param   { yyscan_t scanner }
+%lex-param   { void *scanner }
 
 %initial-action
 {
