diff --git a/flex/stack_description_scanner.l b/flex/stack_description_scanner.l
index d512c05..10d1c21 100644
--- a/flex/stack_description_scanner.l
+++ b/flex/stack_description_scanner.l
@@ -98,8 +98,7 @@ signed_integer          [+-]?[[:digit:]]+
 exponent                [eE]{signed_integer}
 double                  {signed_integer}((\.{unsigned_integer})?{exponent}?)?
 identifier              [[:alpha:]](\_|[[:alnum:]])*
-path_begin              (\.{1,2}\/|\/)?
-path                    \"{path_begin}{identifier}(\/|\.{1,2}|{identifier})*\"
+path                    \"[^"]+\"
 
 /* exclusive start conditions to exclude C/C++ like comments in */
 /* the scanned file.                                            */
